### PR TITLE
chore(alias): fix lint errors

### PR DESCRIPTION
**Pull request by benmccann**
`lint` is failing on `master` for this project. This fixes it

### CODE REVIEW

File: packages/alias/src/index.ts [hunk 1]
```diff
@@ -25,9 +25,7 @@ function matches(pattern: string | RegExp, importee: string) {
   return importeeStartsWithKey && importeeHasSlashAfterKey;
 }
 
-function normalizeId(id: string): string;
-function normalizeId(id: string | undefined): string | undefined;
-function normalizeId(id: string | undefined) {
+function normalizeId<T>(id: T) {
```
Comment (NotWoods) [inline packages/alias/src/index.ts:28]: This should be restricted to just strings
```suggestion
function normalizeId<T extends string | undefined>(id: T) {
```

**Review by NotWoods — CHANGES_REQUESTED** [review]
What lint errors are being triggered here?

**Comment by benmccann** [comment]
>packages/alias/src/index.ts
  29:10  error  'normalizeId' is already defined  no-redeclare
  30:10  error  'normalizeId' is already defined  no-redeclare

File: packages/alias/src/index.ts (hunk 1 above)
Comment (benmccann) [reply packages/alias/src/index.ts]: makes sense. done

**Comment by NotWoods** [comment]
This sounds like an ESLint issue as we're using a standard TypeScript feature here called function overloading. I'd prefer to keep the overloads as its slightly faster and easier to read.

**Comment by benmccann** [comment]
Ok, I've reverted the fix for the `no-redeclare` failure and just included the other fix for now. I can report an issue to the typescript eslint project after this is merged for the remaining issue

**Review by NotWoods — APPROVED** [review]

**Comment by shellscape** [comment]
Closing in favor of #708. The fix for the linting as related to a typescript upgrade (whether or not that's the cause) should be done there since it's related.

**Pull request closed without merging**